### PR TITLE
Add prettier

### PR DESCRIPTION
**Pull request by wdhorton**
And fixes competing lint rules

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +45,8 @@
     "prepack-prepack": "node --stack_size=10000 --max_old_space_size=8096 ./bin/prepack.js ./lib/prepack-cli.js --out ./lib/prepack-cli.prepacked.js --compatibility node-cli --mathRandomSeed rnd",
     "validate": "npm run build && npm run build-scripts && npm run lint && npm run depcheck && npm run flow && npm test",
     "prepublish": "npm run build",
-    "depcheck": "babel-node scripts/detect_bad_deps.js"
+    "depcheck": "babel-node scripts/detect_bad_deps.js",
+    "prettier": "prettier --write --single-quote --jsx-bracket-same-line --trailing-comma all --print-width 120 \"+(src|scripts)/**/*.js\""
```
Comment (NTillmann) [inline package.json]: Is there a way to move all these configurations into a separate configuration file? This command-line is starting to get big.

Comment (NTillmann) [inline package.json]: By far most changes are around turning strings to single-quote. This makes it really difficult to code review the actual changes. Is there a way to leave strings as they are? If not, I vote for starting out with double-quotes, and possibly changing that default later.

Comment (NTillmann) [inline package.json]: Checking for correct formatting should become part of validation and what's checked by  `circle.yml`.

**Comment by NTillmann** [comment]
It looks like it will be tough to land these changes while we keep evolving Prepack...
How about breaking this up:
1. A pull request with just adding the config changes to enable running Prettier, but not actually applying Prettier to all sources. Approving and merging should be quick.
2. Then I can run Prettier once and change the ES-Lint rules, and merge this quickly.
3. Then we can figure out how to enforce Prettier in the circle configuration.

**Comment by wdhorton** [comment]
That makes sense to me!

**Pull request closed without merging**